src/Core/Client.vala

```diff
@@ -277,6 +277,7 @@ public class AppCenterCore.Client : Object {
                 var pk_package = package.find_package ();
                 if (pk_package != null && pk_package.get_info () == Pk.Info.INSTALLED) {
                     package.installed_packages.add (pk_package);
+                    populate_timestamp (package, pk_package, get_install_transactions_sync ());
```

Comment:
Note that calling `get_install_transactions_sync` could be a bottleneck on some systems. Calling it in different places is unnecessary because really it will return the same results as the previous calls and will make AppCenter's backend load time slower than it already is. You should get those transactions only once at the beggining of Client initialization and then use them throughout the Client class.